tests/test_performance_harness.py

```
@@ -74,22 +87,136 @@ def test_host_metrics_include_bounded_network_interface_counters(self):
+        rejected.exception.close()
+
+    def test_loopback_probe_ignores_ambient_proxy_environment(self):
+        proxy_requests = []
+
+        class RecordingProxyHandler(BaseHTTPRequestHandler):
+            def do_GET(self):
+                proxy_requests.append(self.path)
+                self.send_response(200)
+                self.end_headers()
+
+            def log_message(self, format, *args):
```

Ruff (0.16.1)

[error] Function argument `format` is shadowing a Python builtin

(A002)